zebra/zebra_nhg.c

```diff
@@ -1848,15 +1881,49 @@ uint8_t zebra_nhg_nhe2grp(struct nh_grp *grp, struct nhg_hash_entry *nhe,
 			}
 		}
 
-		/* Check for duplicate IDs, kernel doesn't like that */
-		for (int j = 0; j < i; j++) {
-			if (depend->id == grp[j].id)
-				duplicate = true;
-		}
+		if (!zebra_nhg_depends_is_empty(depend)) {
+			/* This is a group within a group */
+			i = zebra_nhg_nhe2grp_internal(grp, i, depend, max_num);
```

Comment:
Oh man - so many things to dislike about recursion... have to check `i` everywhere it _could_ have been changed, before using it to index into `grp`, right? it's only being checked at the very end right now, and that could be too late...

Reply:
Good point. I will see if I can move the check somewhere safer.

Reply:
moved i to top of the loop (was at the very end)